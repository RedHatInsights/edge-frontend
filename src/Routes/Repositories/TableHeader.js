import React from 'react';
import { TextContent, Text, TextVariants } from '@patternfly/react-core';
import { ExternalLinkAltIcon } from '@patternfly/react-icons';

const Header = () => {
  return (
    <TextContent
      data-testid="repositories-table-header"
      style={{ padding: '15px', backgroundColor: 'white' }}
    >
      <Text
        data-testid="repositories-table-header-title"
        component={TextVariants.h1}
      >
        Custom repositories
      </Text>
      <Text
        data-testid="repositories-table-header-description"
        component={TextVariants.p}
      >
        Add custom repositories to build RHEL for Edge images with additional
        packages.{' '}
        <Text style={{ display: 'inline' }} component={TextVariants.small}>
          <Text
            data-testid="repositories-table-header-link"
            component={TextVariants.a}
            href="#"
          >
            Learn more
            <ExternalLinkAltIcon size="sm" className="pf-u-ml-sm" />
          </Text>
        </Text>
      </Text>
    </TextContent>
  );
};

export default Header;
