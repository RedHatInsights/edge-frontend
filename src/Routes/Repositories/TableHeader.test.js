import React from 'react';
import TableHeader from './TableHeader';
import { render, screen } from '@testing-library/react';

describe('Repository table header', () => {
  it('should render correctly', () => {
    render(<TableHeader />);

    const title = screen.getByTestId('repositories-table-header-title');
    const description = screen.getByTestId(
      'repositories-table-header-description'
    );
    const link = screen.getByTestId('repositories-table-header-link');

    expect(title.innerHTML).toEqual('Custom repositories');
    expect(description.firstChild.textContent).toEqual(
      'Add custom repositories to build RHEL for Edge images with additional packages.'
    );
    expect(link.textContent).toEqual('Learn more');
    expect(link.getAttribute('href')).toEqual('#');
  });
});
