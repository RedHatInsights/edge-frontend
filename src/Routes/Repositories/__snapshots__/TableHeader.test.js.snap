// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHeader renders correctly 1`] = `
<div
  class="pf-v5-c-content"
  style="padding: 15px; background-color: white;"
>
  <h1
    class=""
    data-ouia-component-id="OUIA-Generated-Text-1"
    data-ouia-component-type="PF5/Text"
    data-ouia-safe="true"
    data-pf-content="true"
  >
    Custom repositories
  </h1>
  <p
    class=""
    data-ouia-component-id="OUIA-Generated-Text-2"
    data-ouia-component-type="PF5/Text"
    data-ouia-safe="true"
    data-pf-content="true"
  >
    Add custom repositories to build RHEL for Edge images with additional packages.
     
  </p>
</div>
`;
